code/modules/projectiles/ammunition/magazines.dm

```diff
@@ -514,28 +514,26 @@
 	multiple_sprites = 1
 
 
-/obj/item/ammo_magazine/ma12gclip
+/obj/item/ammo_magazine/clip/c12g
 	name = "ammo clip (12g beanbag)"
 	icon_state = "12gclip_bean" //largely a codersprite, looks good enough. feel free to make a better one.
 	desc = "A metal clip for holding and quickly loading shotgun shells. This one is loaded with beanbag shells."
 	caliber = "12g"
 	ammo_type = /obj/item/ammo_casing/a12g/beanbag
-	matter = list(DEFAULT_WALL_MATERIAL = 1580) // beanbag shells x6 + 500 metal for the clip itself.
-	max_ammo = 6
-	multiple_sprites = 1
-	mag_type = SPEEDLOADER
+	matter = list(DEFAULT_WALL_MATERIAL = 1070) // beanbag shells x4 + 350 metal for the clip itself.
+	max_ammo = 4
 
-/obj/item/ammo_magazine/ma12gclip/pellet
+/obj/item/ammo_magazine/clip/c12g/pellet
 	name = "ammo clip (12g buckshot)"
 	icon_state = "12gclip_p"
 	ammo_type = /obj/item/ammo_casing/a12g/pellet
-	matter = list(DEFAULT_WALL_MATERIAL = 2660)
+	matter = list(DEFAULT_WALL_MATERIAL = 1790)
 
-/obj/item/ammo_magazine/ma12gclip/slug
+/obj/item/ammo_magazine/clip/c12g/slug
 	name = "ammo clip (12g slug)"
 	icon_state = "12gclip_s"
 	ammo_type = /obj/item/ammo_casing/a12g
```

Comment:
It's based in slugs because a slug is the most simple type. It's just a regular bullet casing. Nothing more nothing less.

If you do not change this, if I go and decide later I want to add shotgun ammo to a box or something, I know shotgun casings are under the obj/item/ammo_casing/a12g path, so that implies that /obj/item/ammo_magazine/clip/c12g holds the same kind of casing. That is not how you have it written, which is very bad practice.